src/parser.ts

```diff
@@ -174,12 +195,37 @@ export class EnvParser {
   /**
    * Parse the env string to an object of environment variables.
    */
-  parse(): DotenvParseOutput {
+  async parse(): Promise<DotenvParseOutput> {
     const envCollection = dotenv.parse(this.#envContents.trim())
+    const identifiers = Object.keys(EnvParser.#identifiers)
+    let result: DotenvParseOutput = {}
+
+    $keyLoop: for (const key in envCollection) {
```

Comment:
Not sure, I understand this `$keyLoop` syntax. Can't this code be as simple as

```ts
for (const key in envCollection) {
  const value = this.#getValue(key, envCollection)

  if (value.includes(':')) {
    for (const identifier of identifiers) {
      if (value.startsWith(`${identifier}\\:`)) {
        result[key] = identifier + value.substring(identifier.length + 1)
      } else if (value.startsWith(`${identifier}:`)) {
        result[key] = await EnvParser.#identifiers[identifier](
          value.substring(identifier.length + 1)
        )
      }
    }
    result[key] = value
  } else {
    result[key] = value
  }
}
```

Reply:
We have to break the outer loop.

In your code example, you are reassigning the `result[key]` array at the end of the second `for` loop.

```ts
for (const identifier of identifiers) {
  if (value.startsWith(`${identifier}\\:`)) {
    result[key] = identifier + value.substring(identifier.length + 1)
  } else if (value.startsWith(`${identifier}:`)) {
    result[key] = await EnvParser.#identifiers[identifier](
      value.substring(identifier.length + 1)
    )
  }
}
result[key] = value // 👈 Here the value is lost
```

A [JavaScript label](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label) can be used to name a loop and use `continue` or `break` on it even if you are into another loop.

Reply:
I mean, we can use the `break` statement without having the label.

To be honest, I have never used labels and neither they are anywhere in the codebase. Are they usually helpful when you are inside nested loops?

Reply:
> I mean, we can use the break statement without having the label.

We have to `continue` on outside loop, this is the issue.

> I have never used labels and neither they are anywhere in the codebase

If you prefer, we can make it work by storing a temporary variable that defines if the value has been fine and then continue the loop after. Something like:

```ts
for (const key in envCollection) {
  let valueFound = false // 👈
  const value = this.#getValue(key, envCollection)
  
  if (value.includes(':')) { 
    for (const identifier of identifiers) {
      if (value.startsWith(`${identifier}\\:`)) {
        result[key] = identifier + value.substring(identifier.length + 1)
        valueFound = true // 👈
        break
      }
      
      if (value.startsWith(`${identifier}:`)) {
        result[key] = await EnvParser.#identifiers[identifier](
          value.substring(identifier.length + 1)
        )
        valueFound = true // 👈
        break
      }
    }
    
    if (valueFound) { // 👈
      continue
    }
    
    result[key] = value
  } else {
    result[key] = value
  }
}
```

Reply:
Okay, maybe I am missing something. Let's go with the label approach :)